src/domain/entities/PeriodDate.ts

```diff
@@ -0,0 +1,52 @@
+import { Struct } from "$/domain/entities/generic/Struct";
+
+export type PeriodDetailsAttrs = { year: number; startDate: string; endDate: string };
+type PeriodDateAttrs = { startDate: string; endDate: string; periods: PeriodDetailsAttrs[] };
+
+export class PeriodDate extends Struct<PeriodDateAttrs>() {
```

Comment:
In some other comment I had doubts about the naming. Maybe is the order? `DatePeriod` meaning "period of dates" seems more fitting.